tests/test_dynamodb_log_based_parent_child_data.py

```diff
@@ -0,0 +1,99 @@
+from boto3.dynamodb.types import TypeSerializer
+
+from tap_tester import connections
+from tap_tester import menagerie
+from tap_tester import runner
+
+from base import TestDynamoDBBase
+
+class DynamoDBLogBasedParentChildData(TestDynamoDBBase):
+    """
+        Test case for verifying:
+        - The tap does not break when the parent data is not found and the user is requesting for child data
+        - The tap does not break when the data a specific position is not found in the record
```

Comment:
I don't see that both cases are being tested. There should be a multiple logical syncs taking place if we are testing multiple cases like this.